src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -122,7 +122,122 @@ public void pairDevice(
       Log.d(TAG, "Bluetooth connection added with ID: " + connectionId);
       Log.d(TAG, "Pairing device with ID: " + deviceId);
       pairDevice(
-          deviceControllerPtr, deviceId, connectionId, setupPincode, csrNonce, networkCredentials);
+          deviceControllerPtr,
+          deviceId,
+          connectionId,
+          setupPincode,
+          csrNonce,
+          networkCredentials,
+          null,
+          0);
+    } else {
+      Log.e(TAG, "Bluetooth connection already in use.");
+      completionListener.onError(new Exception("Bluetooth connection already in use."));
+    }
+  }
+
+  /**
+   * Pair a device connected through BLE.
+   *
+   * <p>If the completionCallback non-null, when {@link
+   * DeviceAttestationDelegate.DeviceAttestationCompletionCallback#onDeviceAttestationCompleted(long,
+   * long, AttestationInfo, int)} is received, {@link #continueCommissioning(long, boolean)} must be
+   * called.
+   *
+   * @param bleServer the BluetoothGatt representing the BLE connection to the device
+   * @param connId the BluetoothGatt Id representing the BLE connection to the device
+   * @param deviceId the node ID to assign to the device
+   * @param setupPincode the pincode for the device
+   * @param csrNonce the 32-byte CSR nonce to use, or null if we want to use an internally
+   *     randomlygenerated CSR nonce.
+   * @param networkCredentials the credentials (Wi-Fi or Thread) to be provisioned
+   * @param completionCallback the callback will be invoked when deviceattestation completed with
+   *     device info for additional verification.
+   * @param failSafeExpiryTimeout the value to set for the fail-safe timer before
+   *     onDeviceAttestationCompleted is invoked.
+   */
+  public void pairDevice(
+      BluetoothGatt bleServer,
+      int connId,
+      long deviceId,
+      long setupPincode,
+      @Nullable byte[] csrNonce,
+      NetworkCredentials networkCredentials,
+      @Nullable DeviceAttestationDelegate.DeviceAttestationCompletionCallback completionCallback,
+      int failSafeExpiryTimeout) {
```

Comment:
This does not belong in this API. This should be based on setters on the class.

It should be possible to override the entire device attestation flow, rather than special-case on failure. The failure will still occur frequently since there is no way to provide an override of PAA trust store.

Reply:
> This does not belong in this API. This should be based on setters on the class.
> 
> It should be possible to override the entire device attestation flow, rather than special-case on failure. The failure will still occur frequently since there is no way to provide an override of PAA trust store.

Thanks for your conversation, I will fix your requested changes next week. Happy weekend.